tsdb/chunks/head_chunks.go

```diff
@@ -219,13 +219,39 @@ func listChunkFiles(dir string) (map[int]string, error) {
 		return nil, err
 	}
 	res := map[int]string{}
+	lastFile := uint64(0)
 	for _, fi := range files {
 		seq, err := strconv.ParseUint(fi.Name(), 10, 64)
 		if err != nil {
 			continue
 		}
+		if seq > lastFile {
+			lastFile = seq
+		}
 		res[int(seq)] = filepath.Join(dir, fi.Name())
 	}
+
+	if lastFile != uint64(0) {
```

Comment:
Design wise, it looks weird to me doing this reparing in the `listChunkFiles()` which, in my opinion, should just be focused on listing chunk files. Have you considered creating a dedicated function `repairLastChunkFile()` and calling it from `openMMapFiles()` right after `listChunkFiles()`?

Reply:
Splitting it into a dedicated function and calling it elsewhere makes perfect sense. I will make the change.